extensions/amp-fx-collection/0.1/test/integration/test-amp-fx-fly-in.js

```diff
@@ -0,0 +1,167 @@
+  // `data-fly-in-distance` and `data-duration` parameters differ/device.
+  const bodyLeft = `
+    <div class="spacer"></div>
+    <div id="animTarget"
+      amp-fx="fly-in-left"
+      data-duration="500ms"
+      data-fly-in-distance="25%"
+      data-margin-start="20%">
+    </div>
+    <div class="spacer"></div>
+  `;
+
+  describes.integration("amp-fx='fly-in-left'", {
+    body: bodyLeft,
+    css,
+    extensions,
+  }, env => {
+
+    let win;
+    beforeEach(() => {
+      win = env.win;
+      toggleExperiment(win, 'amp-fx-fly-in', true, false);
+    });
+
+    it('runs fly-in-left animation with default parameters', () => {
+      expect(isExperimentOn(win, 'amp-fx-fly-in')).to.be.true;
+      const initialLeft = getComputedLeft(win);
+      win.scrollTo(0, 0.5 * getViewportHeight(win));
+      return Promise.resolve().then(timeout(10000))
+          .then(() => {
+            expect(getComputedLeft(win)).to.be.above(initialLeft);
+          });
+    });
+  });
+
+  // Can't test a default `fly-in-right` animation as the
+  // `data-fly-in-distance` and `data-duration` parameters differ/device.
+  const bodyRight = `
+    <div class="spacer"></div>
+    <div id="animTarget"
+      amp-fx="fly-in-right"
+      data-duration="500ms"
+      data-fly-in-distance="25%"
+      data-margin-start="20%">
+    </div>
+    <div class="spacer"></div>
+  `;
+
+  describes.integration("amp-fx='fly-in-right'", {
+    body: bodyRight,
+    css,
+    extensions,
+  }, env => {
+
+    let win;
+    beforeEach(() => {
+      win = env.win;
+      toggleExperiment(win, 'amp-fx-fly-in', true, false);
+    });
+
+    it('runs fly-in-right animation with default parameters', () => {
+      expect(isExperimentOn(win, 'amp-fx-fly-in')).to.be.true;
+      const initialLeft = getComputedLeft(win);
+      win.scrollTo(0, 0.5 * getViewportHeight(win));
+      return Promise.resolve().then(timeout(10000))
+          .then(() => {
+            expect(getComputedLeft(win)).to.be.below(initialLeft);
+          });
+    });
+  });
+
+  // Can't test a default `fly-in-bottom` animation as the
+  // `data-fly-in-distance` and `data-duration` parameters differ/device.
+  const bodyBottom = `
+    <div class="spacer"></div>
+    <div id="animTarget"
+      amp-fx="fly-in-bottom"
+      data-duration="500ms"
+      data-fly-in-distance="25%"
+      data-margin-start="20%">
+    </div>
+    <div class="spacer"></div>
+  `;
+
+  describes.integration("amp-fx='fly-in-bottom'", {
+    body: bodyBottom,
+    css,
+    extensions,
+  }, env => {
+
+    let win;
+    beforeEach(() => {
+      win = env.win;
+      toggleExperiment(win, 'amp-fx-fly-in', true, false);
+    });
+
+    it('runs fly-in-bottom animation with default parameters', () => {
+      expect(isExperimentOn(win, 'amp-fx-fly-in')).to.be.true;
+      const initialTop = getComputedTop(win);
+      win.scrollTo(0, 0.5 * getViewportHeight(win));
+      return Promise.resolve().then(timeout(2000))
```

Comment:
Why are these timeouts different? If we can get by with 2 seconds, maybe lets not wait 10 seconds in the two tests above. =)

Reply:
Oops. That was a mistake left behind from debugging. Reverted back to `2000ms` now.